index.js

```diff
@@ -69,64 +69,76 @@ async function run() {
     configs.repoOptions
   );
 
-  return octokit.paginate(workflowRunsRequest).then(workflowRuns => {
-    const artifactPromises = workflowRuns
-      .filter(workflowRun => {
-        const skipWorkflow =
-          configs.skipTags && taggedCommits.includes(workflowRun.head_sha);
-
-        if (skipWorkflow) {
-          console.log(`Skipping tagged run ${workflowRun.head_sha}`);
-
-          return false;
-        }
-
-        return true;
-      })
-      .map(workflowRun => {
-        const workflowRunArtifactsRequest = octokit.actions.listWorkflowRunArtifacts.endpoint.merge(
-          {
-            ...configs.repoOptions,
-            run_id: workflowRun.id,
+  return octokit
+    .paginate(workflowRunsRequest, ({ data }, done) => {
+      const stopPagination = data.find(
+        workflowRun => moment.utc().diff(workflowRun.created_at, "days") > 90
```

Comment:
I like the proposed `isBefore` approach better 👍

Reply:
ba978a2